example/anchor.js

```diff
@@ -0,0 +1,172 @@
+// This example shows two ways to anchor content, Direct and Merklized.
+//
+// The first, direct anchoring, places the content directly into a transaction. This method makes
+// the content public and immutable but direct anchors become expensive to post as content becomes
+// large. Direct anchors are also costly to post in bulk (i.e. anchoring multiple documents).
+//
+// It's advised to hash documents in order to keep anchoring price constant irrespective of document size.
+// It's advised to batch document hashes into a merkle tree to keep anchoring price constant
+// irrespective of document count.
+//
+// A potential downside to merkle batching is that PoE verification of each batched document will
+// require a merkle proof of inclusion. This is a tradeoff to consider when designing for your use
+// case. Merkleized anchors can batch infinite anchors into a single, fixed cost transaction, but
+// verification requires a proof.
+//
+// This example uses mrklt for merkle tree construction. A mrklt tree containing exactly one
+// document will have `root = hash(hash(document))` and the merkle proof for that document will be
+// an empty list `[]`. `verify_proof(hash(document), []) = root = hash(hash(document))`. In other
+// words, when batch size is 1, we can infer `proof = []`.
+//
+// It is possible simply post `hash(document)` as an anchor, but it's recommended to double-hash
+// the document instead. Since `hash(hash(document)) = compute_root(hash(document))` a double-hashed
+// anchor can be interpreted as the root of a merkle-tree with 1 leaf. Since the merkle tree has
+// only one leaf, the proof of inclusion for that leaf will be empty.
+
+/* eslint-disable import/no-extraneous-dependencies, import/newline-after-import */
+
+/* eslint-disable camelcase */
+import { compute_root, create_proof, verify_proof } from 'mrklt';
```

Comment:
Can you add the support for batching utils in the SDK source as examples are not available when SDK build is used? Also, can you add a single function to generate tree from given documents, and return root and proofs as thats how this is supposed to be used. Calling `create_proof` for each document (leaf) recreates the tree which becomes expensive when trying to anchor 1000s of documents.

As an enhancement, hash function (for tree) can be abstracted by accepting a hash function/hash interface as an arg. such that callers are not forced to use blake2s (for wasm as well).

Reply:
Added ticket for optimization: https://github.com/docknetwork/mrklt/issues/3
Added ticket for enhancement: https://github.com/docknetwork/mrklt/issues/4